### PR TITLE
PHP 7.4

### PR DESCRIPTION
**Pull request by sam-bee**
To allow API SDK to be used by a wider range of clients, add compatibility with PHP 7.4+

### CODE REVIEW

File: src/Data/AbstractParams.php [hunk 1]
```diff
@@ -13,24 +13,26 @@
...
     ) {
-        //
+        $this->params = $params;
     }
 
     /**
      * Obtain a new instance.
      */
-    public static function new(array $params = []): static
+    public static function new(array $params = []): self
```
Comment (uphlewis) [inline src/Data/AbstractParams.php:27]: With these i think we need an @return docblock to indicate it actually returns static, not self

File: composer.json [hunk 2]
```diff
@@ -18,17 +18,20 @@
...
+        "php": "7.4.* | 8.0.* | 8.1.* | 8.2.* | 8.3.*",
+        "ext-json": "*",
+        "nyholm/psr7": "^1.8",
         "php-http/client-common": "^2.7",
         "php-http/discovery": "^1.19",
         "php-http/logger-plugin": "^1.3",
         "psr/http-client-implementation": "*",
-        "psr/log": "^3.0"
+        "psr/log": "*",
+        "symfony/http-client": "^5.4"
```
Comment (uphlewis) [inline composer.json:29]: How come symfony/http-client was added here?

**Review by uphlewis — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (sam-bee) [reply composer.json:29]: What version of composer did you use to create the file, just out of interest?

Comment (sam-bee) [reply composer.json:29]: I have moved the package to the  `require-dev` section

Comment (uphlewis) [reply composer.json:29]: > What version of composer did you use to create the file, just out of interest?

Probably the one in the php 8.2 dev container - v2.7.7

Comment (uphlewis) [reply composer.json:29]: The objective of the httplug abstraction is that we dont depend on a specific client implementation in the library and leave that up to the consumer - if we need an implementation for tests, there's a mock client we can use in require-dev: https://docs.php-http.org/en/latest/clients/mock-client.html

Comment (sam-bee) [reply composer.json:29]: Following on from conversation on slack, the `symfony/http-client` dependency is being left in the `require-dev` section for now

**Pull request merged**